boot/zephyr/nrf_cleanup.c

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <hal/nrf_rtc.h>
+
+void nrf_cleanup_peripheral(void) {
+#if CONFIG_HAS_HW_NRF_RTC0
+	nrf_rtc_task_trigger(NRF_RTC0, NRF_RTC_TASK_STOP);
+#endif
+#if CONFIG_HAS_HW_NRF_RTC1
+	nrf_rtc_task_trigger(NRF_RTC1, NRF_RTC_TASK_STOP);
+#endif
+#if CONFIG_HAS_HW_NRF_RTC2
+	nrf_rtc_task_trigger(NRF_RTC2, NRF_RTC_TASK_STOP);
+#endif
+#if CONFIG_HAS_HW_NRF_UARTE0
+	nrf_uarte_disable(NRF_UARTE0);
+#endif
+#if CONFIG_HAS_HW_NRF_UARTE1
+	nrf_uarte_disable(NRF_UARTE1);
+#endif
+}
+
+void nrf_cleanup_nvic(void) {
```

Comment:
This could be done for all Cortex-M, right? Not just nrf.

Reply:
https://github.com/JuulLabs-OSS/mcuboot/pull/683